build api: fix release version

0xc0170 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

This is non trivial fix as the function is being used outside of this repository.
Tools rely on it to return list of targets for 2 or 5. As we removed release_version from many targets,
this broke the logic. To keep the logic as it was, without updating all tools out there now,
lets just return full set of targets - all are supported.

In case for Mbed 2, returning all targets does not make sense, but rather raise an exception here. Not supported. This avoids suprised. If you look at build api functions there are many checks for 2 or 5 so more
clean up needed to actually get release_version out of the tools.

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [x] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

0xc0170 committed:
build api: fix release version

This is non trivial fix as the function is being used outside of this repository.
Tools rely on it to return list of targets for 2 or 5. As we removed release_version from many targets,
this broke the logic. To keep the logic as it was, without updating all tools out there now,
lets just return full set of targets - all are supported.

In case for Mbed 2, returning all targets does not make sense, but rather raise an exception here. Not supported. This avoids suprised. If you look at build api functions there are many checks for 2 or 5 so more
clean up needed to actually get release_version out of the tools.

hugueskamba commented on tools/build_api.py:
```diff
@@ -411,26 +411,25 @@ def get_mbed_official_release(version):
+    # in targets.json file
+
+    if version == `2`:
+        raise InvalidReleaseTargetException("Mbed 2 not supported anymore with this version of tools")
```
```suggestion
        raise InvalidReleaseTargetException("Mbed 2 is no longer supported with this version of the tools")
```

hugueskamba commented on tools/build_api.py:
```diff
@@ -411,26 +411,25 @@ def get_mbed_official_release(version):
+    if version == `2`:
+        raise InvalidReleaseTargetException("Mbed 2 not supported anymore with this version of tools")
+
     mbed_official_release = (
```
I am not sure the syntax is correct here. Can you run the script with your changes?

0xc0170 replied:
I was checking what CI expected and seems to be as it is here. will study CI results once we get some, if all is being tested as it should

0xc0170 committed:
tools/build_api: fix wording

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

0xc0170 committed:
build: fix error syntax

0xc0170 committed:
tools build: fix toolchain tuple, should be list

0xc0170 committed:
tools build: return empty list for mbed2

We cant afford to raise an exception now. As anyone using this out there do not catch it. Rather
an empty list, as it can work after this fix - won't do anything.

0xc0170 committed:
RZ_A1XX: public set to false as its base target

jamesbeyond commented on tools/build_api.py:
```diff
@@ -411,26 +411,25 @@ def get_mbed_official_release(version):
                     TARGET_MAP[target].name,
-                    tuple(transform_release_toolchains(
-                        TARGET_MAP[target], version))
+                    tuple(['ARM', 'GCC_ARM'])
```
just wonder why do we use hard-coded tuple here?  instead of calling the function? and IAR been dropped?
We can fix the function if need?

0xc0170 replied:
I wanted to avoid refactoring at the moment (so changes just to one function to get the functionality needed). `transform_release_toolchains` does not have much meaning anymore as only ARM or GCC ARM is supported, so this can be hardcoded here

0xc0170 committed:
tools build: add profile

To select supported_application_profiles attribute from targets

0xc0170 committed:
targets: add rtos profile by default

Every target is assumed to supported Mbed OS. If its not the case, it supports
only baremetal. Thus removing rtos, and adding bare-metal to the app profile.

0xc0170 committed:
targets: enable baremetal by default

For only baremetal targets, just drop in replace. Don't need to add/remove.
